### PR TITLE
Merge Simulation related fixing/improvment into release 8.4.1

### PR DESCRIPTION
**Pull request by shapon2004**

### CODE REVIEW

**Comment by shapon2004** [comment]
AP-6446 Sim parameters: limit total number of instances to be simulated
AP-6447 Discover hourly cost information from event log and use in sim parameters
AP-3821 Change error message of simulation timeframe

File: Apromore-Custom-Plugins/Process-Simulation-Info-Logic/src/main/java/org/apromore/processsimulation/config/SimulationAdditionalParam.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ * The intellectual and technical concepts contained herein are
+ * proprietary to Apromore Pty Ltd and its suppliers and may
+ * be covered by U.S. and Foreign Patents, patents in process,
+ * and are protected by trade secret or copyright law.
+ * Dissemination of this information or reproduction of this
+ * material is strictly forbidden unless prior written permission
+ * is obtained from Apromore Pty Ltd.
+ * #L%
+ */
+
```
Comment (Janite) [inline Apromore-Custom-Plugins/Process-Simulation-Info-Logic/src/main/java/org/apromore/processsimulation/config/SimulationAdditionalParam.java]: This should be the license header for ApromoreCore

File: Apromore-Core-Components/Apromore-Portal/src/main/webapp/openModelInBPMNio.zul [hunk 2]
```diff
@@ -405,6 +405,8 @@
             zk.Widget.$('$slider').setCurpos(newFontSize);
             Apromore.BPMNEditor.app.editor.changeFontSize(newFontSize);
           }
+
+          Apromore.BPMNEditor.qbpProcessMaxLimit = (${arg.qbpProcessMaxLimit})?${arg.qbpProcessMaxLimit}:25000;
```
Comment (Janite) [inline Apromore-Core-Components/Apromore-Portal/src/main/webapp/openModelInBPMNio.zul:409]: Can you also add the change from https://github.com/apromore/ApromoreEE/pull/1702? This fixes an issue where `qbpProcessMaxLimit` is not found when we open models in publish mode.

File: Apromore-Custom-Plugins/Process-Simulation-Info-Logic/src/main/java/org/apromore/processsimulation/model/CostingData.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+ * property of Apromore Pty Ltd and its suppliers, if any.
+ * The intellectual and technical concepts contained herein are
+ * proprietary to Apromore Pty Ltd and its suppliers and may
+ * be covered by U.S. and Foreign Patents, patents in process,
+ * and are protected by trade secret or copyright law.
+ * Dissemination of this information or reproduction of this
+ * material is strictly forbidden unless prior written permission
+ * is obtained from Apromore Pty Ltd.
+ * #L%
+ */
```
Comment (Janite) [inline Apromore-Custom-Plugins/Process-Simulation-Info-Logic/src/main/java/org/apromore/processsimulation/model/CostingData.java]: This should be the license header for ApromoreCore

**Review by Janite — CHANGES_REQUESTED** [review]

**Review by Janite — APPROVED** [review]
Looks good!

**Pull request merged**